src/python/WMComponent/DBS3Buffer/MySQL/NewSubscription.py

```diff
@@ -39,6 +39,8 @@ def _createPhEDExSubBinds(self, datasetID, subscriptionInfo, custodialFlag):
 
         # DeleteFromSource is not supported for move subscriptions
         delete_blocks = None
+        # if None, force it to 0 as per the table schema
+        dataLifetime = subscriptionInfo['DatasetLifetime'] or 0
```

Comment:
I prefer this syntax: `dataLifetime = subscriptionInfo.get('DatasetLifetime', 0)` It is safer since it checks if key exists in a dictionary, while in your case it can fail.